### PR TITLE
Add default book recipe difficulty instead of requiring it to be specified every time

### PR DESCRIPTION
**Pull request by Coolthulhu**
There exist a confusing mechanic where recipes define not just difficulty of the recipe itself, but also difficulty of learning it from a particular book.
In many cases, difficulty of learning the recipe is barely related to difficulty of the book or the recipe.

So here's a proposed new standard:
If you can learn from the book and have skill high enough to use the recipe, you can learn the recipe from the book.
That's it. No confusing special cases.

This PR allows setting it optionally. Later the ability to set a different skill level could be removed. Or possibly kept if there are good reasons to keep this confusing but inconsequential mechanic.

### CODE REVIEW

**Comment by kevingranade** [comment]
No problem with setting a default, but you haven't made a case for removing the ability to customize the level at which a skill is learnable.

As long as level to execute a recipe has side effects related to e.g. success chance and skill gain, level to execute and level to learn need to stay decoupled. There are any number of tasks that are difficult to learn, but once learned are trivial to perform, and the easiest way to model that is setting the level to learn higher than the level to execute.

**Comment by Coolthulhu** [comment]
> No problem with setting a default, but you haven't made a case for removing the ability to customize the level at which a skill is learnable.

The problem with that is that it's hard to make a good argument for it.
There are 3 separate cases:
* If you don't have a skill level high enough to learn from book, it explicitly says that you can't understand the language the book is written in. Then you shouldn't be able to understand the recipes in it either.
* When you can learn the recipe earlier than usual, you still need the skill level to actually execute said recipe. Any attempt below that has chance of failure high enough that it's basically trial and error. This is the best case, though - it actually makes sense to try things you barely understand, because if you somehow succeed, you have a high chance of understanding them.
* The last case - the most confusing one - is when you can understand a book, but not recipes in it. Described in detail below.

> There are any number of tasks that are difficult to learn, but once learned are trivial to perform, and the easiest way to model that is setting the level to learn higher than the level to execute.

This modeling is still very bad. In current implementation:
* You don't actually learn things until you try them. You can't try until they're unlocked. This means that some books have recipes you can't try to follow despite reading the whole book.
* Nothing indicates when you will unlock a given book recipe. It's an arcane internal number.
* Once you gain the high skill level, you trivially learn the thing, most likely without trying it once.
* The recipe isn't any harder the first time. The only thing preventing you from understanding is that you can't try.

At this point it doesn't represent what it is supposed to. It is just an arbitrary hard lock on some recipes.
Now, there is nothing wrong with arbitrary locks per se, but this one also happens to be confusing, not indicated anywhere, hard to indicate due to code structure.
It's a partial implementation that ends up causing more trouble than anything else.

I guess the "easier with book" recipes could be kept.
Possibly made more standardized? All autolearned recipes are 1 level easier with book or something like that.

**Comment by ituluwituluwzev** [comment]
Learning from The Historic Weaponsmith is easier than from Ye Scots Cook Beuk, because there is recipe in the former, but only drawing in the latter.

**Comment by kevingranade** [comment]
> If you don't have a skill level high enough to learn from book, it explicitly says that you can't understand the language the book is written in. Then you shouldn't be able to understand the recipes in it either.

This isn't an intrinsic part of the system, it's a random bit of verbiage.  It can simply be changed to "You can't understand most of the material in this book, but you might be able to learn some recipes."  Books can have a mix of content that requires different skill levls to understand, in which case an "advanced" book has a use for a beginner to learn the recipes, and later it can be used for skill study.

> When you can learn the recipe earlier than usual ...

You didn't make it clear how this is a problem, you can sometimes learn the recipe earlier than usual... and?

> This means that some books have recipes you can't try to follow despite reading the whole book.

Working as intended.

> Nothing indicates when you will unlock a given book recipe. It's an arcane internal number.

Sounds like it's potentially worth fixing, but again this is an implementation detail, not an issue with the system as a whole.

> Once you gain the high skill level, you trivially learn the thing

Again, working as intended.

> The recipe isn't any harder the first time. The only thing preventing you from understanding is that you can't try.

The only thing preventing you from understanding is thet it has a requirement too high for you, it's no more arbitrary than any other requirement, and again this is working as intended.

> At this point it doesn't represent what it is supposed to. It is just an arbitrary hard lock on some recipes.

It's precisely supposed to be a hard lock on some recipes, or a relaxation of a hard lock for instances where the difficulty is lower than usual.

> It's a partial implementation that ends up causing more trouble than anything else.

In short you have some issues with the implementation, but instead of outlining what the problems are or proposing fixes, you just want to throw the feature out and replace it with an extremely simplistic alternative.

So the only issues that actually hold up are:
Book description is misleading in cases where recipes have a lower skill requirement than the book itself.
Books don't reveal to the player when recipes will be available for use.
